_posts/2019-10-23-gpt-2/gpt2.Rmd

```diff
@@ -0,0 +1,317 @@
+Himalayas, Andes, Rockies, Alaska and Africa.
+
+Satellite observations reveal that the amount of spring snow cover in the Northern
+Hemisphere has decreased over the past five decades and that the snow is melting
+earlier.
+
+Global sea level rose about 8 inches in the last century. The rate in the last two
+decades, however, is nearly double that of the last century and is accelerating
+slightly every year.
+
+Both the extent and thickness of Arctic sea ice has declined rapidly over the last
+several decades.
+
+The number of record high temperature events in the United States has been
+increasing, while the number of record low temperature events has been decreasing,
+since 1950. The U.S. has also witnessed increasing numbers of intense rainfall events.
+
+Since the beginning of the Industrial Revolution, the acidity of surface ocean
+waters has increased by about 30 percent.13,14 This increase is the result of humans
+emitting more carbon dioxide into the atmosphere and hence more being absorbed into
+the oceans. The amount of carbon dioxide absorbed by the upper layer of the oceans
+is increasing by about 2 billion tons per year.
+
+TL;DR:
+```
+
+```{r}
+gpt2(prompt = text,
+     model = "774M",
+     total_tokens = 100,
+     top_k = 2)
+```
+
+Here is the generated result, whose quality on purpose we don't comment on. (Of course one can't help having "gut reactions"; but to actually present an evaluation we'd want to conduct a systematic experiment, varying not only input prompts but also function parameters. All we want to show in this post is how you can set up such experiments yourself.)
+
+```
+"\nGlobal temperatures are rising, but the rate of warming has been accelerating.
+\n\nThe oceans have absorbed much of the increased heat, with the top 700 meters of
+ocean showing warming of more than 0.4 degrees Fahrenheit since 1969.
+\n\nGlaciers are retreating almost everywhere around the world, including in the
+Alps, Himalayas, Andes, Rockies, Alaska and Africa.
+\n\nSatellite observations reveal that the amount of spring snow cover in the
+Northern Hemisphere has decreased over the past"
+```
+
+Speaking of parameters to vary, -- they fall into two classes, in a way. It is unproblematic to vary the sampling strategy, let alone the prompt. But for tasks like summarization, or the ones we'll see below, it doesn't feel right to have to tell the model how many tokens to generate. Finding the right length of the answer seems to be part of the task. ^[Formally, `total_tokens` isn't a required parameter. If not passed, a default based on model size will be applied, resulting in lengthy output that definitely will have to be processed by some human-made rule.] Breaking our "we don't judge" rule just a single time, we can't help but remark that even in less clear-cut tasks, language generation models that are meant to approach human-level competence would have to fulfill a criterion of _relevance_ [@grice1975logic].
```

Comment:
Really like that you define a criterion here.

Reply:
Grice is from a _very_ distant past, for me, but the point  seemed _relevant_ ;-)